controller/eks-cluster-config-handler.go

```diff
@@ -807,7 +810,7 @@ func (h *Handler) waitForCreationComplete(config *eksv1.EKSClusterConfig, eksSer
 }
 
 // buildUpstreamClusterState
-func BuildUpstreamClusterState(name, managedTemplateID string, clusterState *eks.DescribeClusterOutput, nodeGroupStates []*eks.DescribeNodegroupOutput, ec2Service *ec2.EC2, includeManagedLaunchTemplate bool) (*eksv1.EKSClusterConfigSpec, string, error) {
+func BuildUpstreamClusterState(name, managedTemplateID string, generatedNodeRoleID string, clusterState *eks.DescribeClusterOutput, nodeGroupStates []*eks.DescribeNodegroupOutput, ec2Service *ec2.EC2, includeManagedLaunchTemplate bool) (*eksv1.EKSClusterConfigSpec, string, error) {
```

Comment:
`generatedNodeRoleID` was added to the function signature, but I don't see it being used anywhere. Is it still needed?

Reply:
I think other than this, and the deep-copy conversation above, this lgtm

Reply:
This is an interesting question. Technically no, `generatedNodeRoleID` is not needed. It works how it is now. But I realize it would be best practice for rancher to pass that field in from the Status since it's a Rancher-generated field. Pulling it from EKS implies its EKS-generated. This is really only something devs care about because either way the functionality will be the same.

Reply:
I guess I'm asking why we need to pull it in at all if we don't use it. Is there a valid reason why we may want it in the future and why adding it now would be beneficial?